serve/graph/schema/filter/transaction_filter.graphql

```diff
@@ -0,0 +1,240 @@
+"""
+Filters for querying Transactions within specified criteria related to their execution and placement within Blocks.
+"""
+input TransactionFilter {
+  """
+  Minimum block height from which to start fetching Transactions, inclusive. Aids in scoping the search to recent Transactions.
+  """
+  from_block_height: Int
+
+  """
+  Maximum block height up to which Transactions should be fetched, exclusive. Helps in limiting the search to older Transactions.
+  """
+  to_block_height: Int
+
+  """
+  Minimum Transaction index from which to start fetching, inclusive. Facilitates ordering in Transaction queries.
+  """
+  from_index: Int
+
+  """
+  Maximum Transaction index up to which to fetch, exclusive. Ensures a limit on the ordering range for Transaction queries.
+  """
+  to_index: Int
+
+  """
+  Minimum `gas_wanted` value to filter Transactions by, inclusive. Filters Transactions based on the minimum computational effort declared.
+  """
+  from_gas_wanted: Int
+
+  """
+  Maximum `gas_wanted` value for filtering Transactions, exclusive. Limits Transactions based on the declared computational effort.
+  """
+  to_gas_wanted: Int
+
+  """
+  Minimum `gas_used` value to filter Transactions by, inclusive. Selects Transactions based on the minimum computational effort actually used.
+  """
+  from_gas_used: Int
+
+  """
+  Maximum `gas_used` value for filtering Transactions, exclusive. Refines selection based on the computational effort actually consumed.
+  """
+  to_gas_used: Int
+
+  """
+  Hash from Transaction content in base64 encoding. If this filter is used, any other filter will be ignored.
+  """
+  hash: String
+
+  """
+  Transaction's message to filter Transactions.
+  """
+  message: TransactionMessageInput
```

Comment:
I felt that the name filter was closer to a feature.

So the higher-level input models, TransactionFilter, BlockFilter, etc. felt like models for functionality, 
while the lower-level input models felt like simple input data models rather than filters.